app/controllers/application_controller.rb

```diff
@@ -1,8 +1,9 @@
 class ApplicationController < ActionController::Base
   helper_method :current_user
-
+  
   def current_user
-    @_current_user ||= User.find(session[:user_id]) if session[:user_id]
+    if session[:user_id]
```

Comment:
I might have you walk me through the memoization here. I'm not sure I understand the conditional of session[:user_id]